src/MicrosoftAi/AbstractionMapper.cs

```diff
@@ -0,0 +1,321 @@
+			ModelId = response.Model ?? request.Model,
+			RawRepresentation = response,
+			Usage = ParseOllamaChatResponseUsage(response)
+		};
+
+		return completion;
+	}
+
+	/// <summary>
+	/// Converts Microsoft.Extensions.AI messages and options to an OllamaSharp chat request.
+	/// </summary>
+	/// <param name="apiClient">The API client used for communication.</param>
+	/// <param name="chatMessages">A list of chat messages.</param>
+	/// <param name="options">Optional chat options to configure the request.</param>
+	/// <param name="stream">Indicates if the request should be streamed.</param>
+	public static ChatRequest ToOllamaSharpChatRequest(IOllamaApiClient apiClient, IList<ChatMessage> chatMessages, ChatOptions? options, bool stream)
+	{
+		// unused ChatOptions properties
+		// options.MaxOutputTokens,
+		// options.ToolMode,
+
+		return new ChatRequest
+		{
+			Format = options?.ResponseFormat == ChatResponseFormat.Json ? "json" : null,
+			KeepAlive = null,
+			Messages = ToOllamaSharpMessages(chatMessages),
+			Model = options?.ModelId ?? apiClient.SelectedModel,
+			Options = new Models.RequestOptions
+			{
+				FrequencyPenalty = options?.FrequencyPenalty,
+				PresencePenalty = options?.PresencePenalty,
+				Stop = options?.StopSequences?.ToArray(),
+				Temperature = options?.Temperature,
+				TopP = options?.TopP,
+			},
+			Stream = stream,
+			Template = null,
+			Tools = ToOllamaSharpTools(options?.Tools)
+		};
+	}
+
+	/// <summary>
+	/// Converts a collection of Microsoft.Extensions.AI.<see cref="AITool"/> to a collection of OllamaSharp tools.
+	/// </summary>
+	/// <param name="tools">The tools to convert.</param>
+	private static IEnumerable<Tool>? ToOllamaSharpTools(IEnumerable<AITool>? tools)
+	{
+		return tools?.Select(ToOllamaSharpTool)
+					 .Where(t => t is not null)
+					 .Cast<Tool>();
+	}
+
+	/// <summary>
+	/// Converts an Microsoft.Extensions.AI.<see cref="AITool"/> to an OllamaSharp tool.
+	/// </summary>
+	/// <param name="tool">The tool to convert.</param>
+	private static Tool? ToOllamaSharpTool(AITool tool)
+	{
+		if (tool is AIFunction f)
+			return ToOllamaSharpTool(f.Metadata);
+
+		return null;
+	}
+
+	/// <summary>
+	/// Converts <see cref="AIFunctionMetadata"/> to a <see cref="Tool"/>.
+	/// </summary>
+	/// <param name="functionMetadata">The function metadata to convert.</param>
+	private static Tool? ToOllamaSharpTool(AIFunctionMetadata functionMetadata)
+	{
+		return new Tool
+		{
+			Function = new Function
+			{
+				Description = functionMetadata.Description,
+				Name = functionMetadata.Name,
+				Parameters = new Parameters
+				{
+					Properties = functionMetadata.Parameters.ToDictionary(p => p.Name, p => new Properties
+					{
+						Description = p.Description,
+						Enum = [], // TODO is there such as possible values in AIFunctionParameterMetadata?
+						Type = ToFunctionTypeString(p.ParameterType)
```

Comment:
As above, if you want to be totally precise and get the JSON value type of the parameter, it's possible to read it by testing if `Schema` is `JsonElement`, and if so, casting it and reading the JSON value type.

But this is not super easy and you might find it's sufficient just to not specify any `Type` for now, at least until stronger use cases emerge to justify it.